clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1853,6 +1824,40 @@ private void subscribeInternal(Collection<String> topics, Optional<ConsumerRebal
         }
     }
 
+    /**
+     * Process the events—if any—that were produced by the {@link ConsumerNetworkThread network thread}.
+     * It is possible that {@link ErrorEvent an error}
+     * could occur when processing the events. In such cases, the processor will take a reference to the first
+     * error, continue to process the remaining events, and then throw the first error that occurred.
+     */
+    private boolean process(EventProcessor<BackgroundEvent> processor) {
```

Comment:
what about renaming this to be explicit about what we process here? It gets confusing given that at this consumer level we're dealing with app events and background events

Reply:
I renamed `process()` as `processBackgroundEvents()`. Is that OK?